services/datamanager/builtin/builtin.go

```diff
@@ -362,29 +375,35 @@ func (svc *builtIn) Sync(_ context.Context, _ map[string]interface{}) error {
 	return nil
 }
 
-// Update updates the data manager service when the config has changed.
-func (svc *builtIn) Update(ctx context.Context, cfg *config.Config) error {
+// Reconfigure updates the data manager service when the config has changed.
+func (svc *builtIn) Reconfigure(
+	ctx context.Context,
+	deps resource.Dependencies,
+	conf resource.Config,
+) error {
 	svc.lock.Lock()
 	defer svc.lock.Unlock()
-	svc.cfg = cfg
-
-	svcConfig, ok, err := getServiceConfig(cfg)
-	// Service is not in the config, has been removed from it, or is incorrectly formatted in the config.
-	// Close any collectors.
-	if !ok {
-		svc.closeCollectors()
-		svc.closeSyncer()
+	svcConfig, err := resource.NativeConfig[*Config](conf)
+	if err != nil {
```

Comment:
Previously if we ran into errors grabbing the service config, we shut down any existing collectors/the syncer. It looks like now we just return an error, and if any previously configured collectors/the syncer are running, they're left alone. 

2 questions:
Am I understanding that correctly?
If so, was this intentional?